src/main/kubectl.ts

```diff
@@ -32,70 +33,164 @@ const packageMirrors: Map<string, string> = new Map([
+    return;
   }
 
-  constructor(clusterVersion: string) {
-    const versionParts = /^v?(\d+\.\d+)(.*)/.exec(clusterVersion)
-    const minorVersion = versionParts[1]
-    /* minorVersion is the first two digits of kube server version
-       if the version map includes that, use that version, if not, fallback to the exact x.y.z of kube version */
-    if (kubectlMap.has(minorVersion)) {
-      this.kubectlVersion = kubectlMap.get(minorVersion)
-      logger.debug("Set kubectl version " + this.kubectlVersion + " for cluster version " + clusterVersion + " using version map")
-    } else {
-      this.kubectlVersion = versionParts[1] + versionParts[2]
-      logger.debug("Set kubectl version " + this.kubectlVersion + " for cluster version " + clusterVersion + " using fallback")
-    }
+  const zshScript =
+    `${initScriptVersionString}
+tempkubeconfig=\"$KUBECONFIG\"
 
-    let arch = null
+# restore previous ZDOTDIR
+export ZDOTDIR="$OLD_ZDOTDIR"
 
-    if (process.arch == "x64") {
-      arch = "amd64"
-    } else if (process.arch == "x86" || process.arch == "ia32") {
-      arch = "386"
-    } else {
-      arch = process.arch
-    }
+# source all the files
+"test -f "$OLD_ZDOTDIR/.zshenv" && . "$OLD_ZDOTDIR/.zshenv"
+test -f "$OLD_ZDOTDIR/.zprofile" && . "$OLD_ZDOTDIR/.zprofile"
+test -f "$OLD_ZDOTDIR/.zlogin" && . "$OLD_ZDOTDIR/.zlogin"
+test -f "$OLD_ZDOTDIR/.zshrc" && . "$OLD_ZDOTDIR/.zshrc"
 
-    const platformName = isWindows ? "windows" : process.platform
-    const binaryName = isWindows ? "kubectl.exe" : "kubectl"
+# voodoo to replace any previous occurences of kubectl path in the PATH
+kubectlpath="${this.dirname}"
+helmpath="${helmPath}"
+p=":$kubectlpath:"
+d=":$PATH:"
+d=\${d//$p/:}
+d=\${ d /#: /}
+export PATH="$kubectlpath:$helmpath:\${d/%:/}"
+export KUBECONFIG ="$tempkubeconfig"
+unset tempkubeconfig
+unset OLD_ZDOTDIR
+`;
 
-    this.url = `${this.getDownloadMirror()}/v${this.kubectlVersion}/bin/${platformName}/${arch}/${binaryName}`
+  return fs.promises.writeFile(zshScriptPath, zshScript, { mode: 0o644 })
+}
 
-    this.dirname = path.normalize(path.join(Kubectl.kubectlDir, this.kubectlVersion))
-    this.path = path.join(this.dirname, binaryName)
+function getDownloadMirror() {
+  // MacOS packages are only available from default
+  return packageMirrors.get(userStore.preferences.downloadMirror)
+    || packageMirrors.get("default");
+}
+
+interface ClientOnlyKubeVersion {
+  clientVersion: {
+    major: string;
+    minor: string;
+    gitVersion: string;
+    gitCommit: string;
+    gitTreeState: string;
+    buildDate: string;
+    goVersion: string;
+    compiler: string;
+    platform: string;
+  }
+}
+
+export class Kubectl {
+  public kubectlVersion: string
+
+  public readonly url: string
+  public readonly path: string
+  public get dirname() {
+    return path.dirname(this.path);
+  }
+
+  public static readonly defaultDirectory = path.join((app || remote.app).getPath("userData"), "binaries", "kubectl")
```

Comment:
With this change, path to kubectl binary will point to `~/Library/Application Support/Lens/binaries/` instead of `Library/Application Support/LensDev/binaries/` on dev env. I suggest to revert this back to:

```typescript
static get kubectlDir() {
    return path.join((app || remote.app).getPath("userData"), "binaries", "kubectl")
  }
```

Reply:
Maybe I don't quite understand, but I fail to see how there is a difference between calling this method once (when the package is first imported) and many times (each time it is accessed). Why would `getPath` return a different result? Looking at the docs on the electron website doesn't imply that there would be a difference.

Reply:
Maybe the value for static property is assigned when importing the file time and not everything is set properly on that phase?

Reply:
Perhaps, do you have reason to believe that electron won't be correctly setup by the time we get to buisness code?

Reply:
You can try to move in `src/main/index.ts`
```
const workingDir = path.join(app.getPath("appData"), appName);
app.setName(appName);
if(!process.env.CICD) {
  app.setPath("userData", workingDir);
}
```
right after `import { app, dialog } from "electron"`. Does it make any difference then?

Reply:
Having added `console.log` to the three places that seem to import `Kubectl` as well as just after those lines and they all output the same